File: Lab(ICE-2202)CPP/lab-02.cpp
#include <iostream>
#include <vector>
using namespace std;

int linear_search(vector<int> &arr, int target)
{
    for (int i = 0; i < arr.size(); ++i)
    {
        if (arr[i] == target)
        {
            return i;
        }
    }
    return -1; // Return -1 if element not found
}

int main()
{
    int n;
    cout << "Enter the size of the array: ";
    cin >> n;

    vector<int> arr(n);
    cout << "Enter the elements of the array: ";
    for (int i = 0; i < n; ++i)
    {
        cin >> arr[i];
    }

    int target;
    cout << "Enter the element to search for: ";
    cin >> target;

    int index = linear_search(arr, target);

    if (index != -1)
    {
        cout << "Element " << target << " found at index " << index << endl;
    }
    else
    {
        cout << "Element " << target << " not found in the array." << endl;
    }

    return 0;
}

File: lab-02.py
def linear_search(arr, target):
    """
    Perform linear search to find the target element in the array.
    """
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1


# Input section to take a list and the target element from the user
arr = list(map(int, input("Enter a list of numbers separated by space: ").split()))
target = int(input("Enter the element to search for: "))

# Perform linear search
index = linear_search(arr, target)

# Output the result
if index != -1:
    print("Element", target, "found at index:", index)
else:
    print("Element", target, "not found in the list.")
